core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -466,6 +466,12 @@ public void unparseCall(SqlWriter writer, SqlCall call, int leftPrec,
     }
   }
 
+  public void unparseBoolLiteral(SqlWriter writer,
+      SqlBoolLiteral literal, int leftPrec, int rightPrec) {
+    writer.keyword(
+        literal.getValue() == null ? "UNKNOWN" : (Boolean) literal.getValue() ? "TRUE" : "FALSE");
```

Comment:
if you use a separate class you can probably add a method getBooleanValue in SqlBooleanLiteral